server/src/controllers/mail/getMessageThread.ts

```
@@ -0,0 +1,60 @@
+import { Status } from "../../enums/StatusCodes";
+import { debugClientErr } from "../../errors/errors";
+import { User } from "../../models/user.model";
+import { KoaController } from "../../utils/KoaController";
+
+import { ORMContext } from "../../server";
+import { Message } from "../../models/message.model";
+import { createDictionary } from "../../utils/createDictionary";
+import { GetMessageSchema } from "./zod/GetMessageSchema";
+import { logging } from "../../utils/logger";
+
+/**
+ * Controller to get multiple messages with single threadid for MailBox.
+ *
+ * This controller get single thread
+ *
+ * @param {Context} ctx - The Koa context object, which includes the request body.
+ * @returns {Promise<void>} - A promise that resolves when the controller is complete.
+ * @throws {Error} - Throws an error if the request body is missing required fields or if logging fails.
+ */
+export const getMessageThread: KoaController = async (ctx) => {
+  try {
+    const user: User = ctx.authUser;
+    const { threadid } = GetMessageSchema.parse(ctx.request.body);
+    if (!threadid) {
```

When we don't have something that we expect we should use one of our client safe custom errors that we have in `errors.ts`, instead we should throw an error e.g. `throw permissionErr()` - in your case, you might want to create a new one.

---

the client only expect error field and not showing any message, so if it's error, I has also added errorLog for server